dataset_loaders/videos/davis2017.py

```diff
@@ -0,0 +1,269 @@
+                                             dataset_version)
+
+        self.image_path = os.path.join(self.path,
+                                       'JPEGImages', '480p')
+        self.mask_path = os.path.join(self.path,
+                                      'Annotations', '480p')
+
+        # Prepare data paths
+        if 'train' in self.which_set or 'val' in self.which_set:
+            self.split = split
+        elif 'test-dev' in self.which_set:
+            self.split = 1.
+        else:
+            raise RuntimeError('Unknown set')
+
+        if 'test-dev' in self.which_set and not self.multiobject:
+            raise RuntimeError('Single object instance is not '
+                               'available for test-dev in Davis2017')
+
+        super(Davis2017Dataset, self).__init__(*args, **kwargs)
+
+    def get_names(self):
+        """Return a dict of names, per prefix/subset."""
+        per_video_names = {}
+
+        # Populate self.filenames and self.prefix_list
+        filenames = self.filenames
+        prefix_list = self.prefix_list
+
+        # cycle through the different videos
+        for prefix in prefix_list:
+            exp_prefix = prefix + '/'
+            per_video_names[prefix] = [el.lstrip(exp_prefix) for el in
+                                       filenames if el.startswith(exp_prefix)]
+        return per_video_names
+
+    def load_sequence(self, sequence):
+        """Load a sequence of images/frames
+
+        Auxiliary function that loads a sequence of frames with
+        the corresponding ground truth and their filenames.
+        Returns a dict with the images in [0, 1], their corresponding
+        labels, their subset (i.e. category, clip, prefix) and their
+        filenames.
+        """
+        from skimage import io
+
+        X = []
+        Y = []
+        F = []
+
+        rgbs = self.unique_rgbs
+
+        for prefix, frame_name in sequence:
+            frame = prefix + '/' + frame_name
+
+            img = io.imread(os.path.join(self.image_path, frame + 'jpg'))
+            img = img.astype(floatX) / 255.
+
+            if self.which_set == 'train' or self.which_set == 'valid':
+                mask = cv2.imread(os.path.join(self.mask_path, frame + 'png'))
+                # Convert mask from RGB to ids format
+                _id = 0
+                temp_mask = np.zeros(mask.shape[:-1], np.int32)
+                for rgb in sorted(rgbs[prefix]):
+                    temp_mask[np.where(np.all(mask == rgb,
+                                              -1).astype(int) == 1)] = _id
+                    if self.multiobject:
+                        _id += 1
+                    else:
+                        _id = 1
+
+                Y.append(temp_mask)
+
+            X.append(img)
+            F.append(frame_name + 'jpg')
+
+        if self.which_set == 'test-dev':
+            frame = [el for el in self.filenames if prefix in el][0]
```

Comment:
Again, this would benefit from using a dictionary to store the filenames per prefix in self.filenames